src/environ/_environ_config.py

```diff
@@ -174,11 +194,73 @@ class Sub:
 
     You can nest your configuration as deeply as you wish.
     """
+    default = None if optional else RAISE
     return attr.ib(
-        default=None, metadata={CNF_KEY: _ConfigEntry(None, None, cls, True)}
+        default=default,
+        metadata={CNF_KEY: _ConfigEntry(None, default, cls, True)},
     )
 
 
+def __default_getter(environ, metadata, prefix, name):
+    ce = metadata[CNF_KEY]
+    if ce.name is not None:
+        var = ce.name
+    else:
+        var = "_".join(prefix + (name,)).upper()
+    log.debug("looking for env var '%s'.", var)
+    try:
+        return environ[var]
+    except KeyError:
+        raise MissingEnvValueError(var)
+
+
+def __to_config_recurse(config_cls, environ, prefixes, default=RAISE):
```

Comment:
Same as above.

Reply:
Done, I'll let you resolve once you have a chance to look at the docstring